conan/tools/cmake/cmake.py

```diff
@@ -66,17 +66,16 @@ def configure(self, source_folder=None):
         if not self._conanfile.should_configure:
             return
 
-        source = self._conanfile.source_folder
-        if source_folder:
-            source = os.path.join(self._conanfile.source_folder, source_folder)
-
         build_folder = self._conanfile.build_folder
+        install_folder = self._conanfile.install_folder
+        source = os.path.join(self._conanfile.source_folder, source_folder) \
```

Comment:
Maybe it makes sense to completely remove the ``source_folder`` argument, and have it only defined in the layout? Why to have it distributed? Unless it is explicitly the ``CMakeLists.txt`` location folder within the source one.

Reply:
Exactly, that `source_folder` makes no sense anymore.